File: elixir_graphql_mysql/lib/elixir_graphql_mysql_web/graphql/queries/sales_query.ex
# lib/elixir_graphql_mysql_web/graphql/queries/sales_query.ex
defmodule ElixirGraphqlMysqlWeb.Graphql.Queries.SalesQuery do
  use Absinthe.Schema.Notation

  object :sales_queries do
    @desc "Get all sales data"
    field :all_sales, list_of(:sale) do
      resolve &ElixirGraphqlMysqlWeb.Resolvers.Charts.list_sales/3
    end
  end
end


#======REQUEST========
#query GetSales{
#    allSales{
#        id
#        salesamount
#        salesdate
#    }
#}

File: vite-react/src/graphql/sales_query.ts
import { gql } from '@apollo/client';
  
export const SALES_QUERY = gql`
  query SaleList {
    allSales{
      salesamount
      salesdate
    }
  }
`;

export interface SaleData {
    salesamount: number
    salesdate: string | number
}


export interface SalesListData {
    allSales: SaleData[]  
}






